### PR TITLE
Scalajs support

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds support for Scala.js, note that this implies a relatively large reorganization of the code but only a few changes to the code itself:

* Added a local implementation of `scala.Predef.valueOf` as this is not (yet) supported by scala.js see typelevel/scala#164
* A couple of tests don't work properly on Scala.js due to the way numbers are handled
* Disabled scoverage on scala.js as it is not well supported

### CODE REVIEW

File: js/src/main/scala/valueof/package.scala [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package object valueof {
+  @inline def valueOf[T](implicit vt: ValueOf[T]) = vt.value
```
Comment (cquiroz) [inline js/src/main/scala/valueof/package.scala]: This is slightly hacky but it is basically the implementation of `scala.Predef.valueOf`

**Comment by fthomas** [comment]
LGTM, thanks @cquiroz!

@soronpo If you don't have any objections, I'm going to merge this.

File: shared/src/test/scala/singleton/twoface/TwoFaceDoubleSpec.scala [hunk 2]
```diff
@@ -275,7 +275,8 @@ class TwoFaceDoubleSpec extends Properties("TwoFace.Double") {
   property("Safe toFloat") = verifyTF(TwoFace.Double(1.0).toFloat, 1.0f)
   property("Unsafe toFloat") = verifyTF(TwoFace.Double(us(1.0)).toFloat, us(1.0f))
   property("Safe toString") = verifyTF(TwoFace.Double(1.0).toString, "1.0")
-  property("Unsafe toString") = verifyTF(TwoFace.Double(us(1.0)).toString, us("1.0"))
+  // scala JVM and scala.js differ on how to print doubles without decimals
+  property("Unsafe toString") = verifyTF(TwoFace.Double(us(1.0)).toString, us("1.0")) || verifyTF(TwoFace.Double(us(1.0)).toString, us("1"))
```
Comment (cquiroz) [inline shared/src/test/scala/singleton/twoface/TwoFaceDoubleSpec.scala]: floating point formatting in scala.js is a bit iffy, but 1.5 will do the trick

Comment (soronpo) [inline shared/src/test/scala/singleton/twoface/TwoFaceDoubleSpec.scala]: That will be fine 👍

File: shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala [hunk 3]
```diff
@@ -11,5 +11,5 @@ class ImplicitConversionSpec extends Properties("ImplicitConversion") {
   property("Wrong OpToOp") = {illTyped("""val ret : 2 + 1 = implicitly[4 - 2]"""); true}
   property("Wrong OpToSingleton") = {illTyped("""val ret : 3 = implicitly[4 - 2]"""); true}
   property("Wrong SingletonToOp") = {illTyped("""val ret : 2 + 1 = 4"""); true}
-  property("ValueOf[Op]") = wellTyped {val ret : 3 = valueOf[4 - 1]}
-}
\ No newline at end of file
+  property("ValueOf[Op]") = wellTyped {val ret : 3 = valueof.valueOf[4 - 1]}
```
Comment (cquiroz) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: I'm not sure what you mean. What's LBS? Is this related to #33 ?

Comment (soronpo) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: LBS is Lightbend Scala (the main scala branch). It too does not have `valueOf`, so I removed the dependency in #33 . You can choose to remove `ValueOf` completely from the code and tests, or I will remove it (and the scala-js workaround) after this PR is merged. Do as you wish. After #33, it will be gone.

Comment (cquiroz) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: Ok, Maybe i should wait for #33 to be merged. This PR changes the structure of the dirs to support the scala-js workaround, if we don't need it then we can go back to the original structure

Comment (soronpo) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: I don't want to burden you with the conflicts. I can merge #33 right now. It's ready. 
If most of the changes were done to satisfy `valueOf`, then indeed we should get rid of it before. 
Your choice.

Comment (soronpo) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: Let me suggest another thing. You can pull #33 ' s branch to a new branch locally, and try to apply the changed that make scala-js work. If everything is OK, then let me know and I will merge #33. If you encounter difficulties, then we apply your PR first, and I will handle the conflicts later.

Comment (cquiroz) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: Sounds good but I won't be able to do it until tomorrow. If you need to merge #33 now feel free to do so and i'll rebase later

Comment (soronpo) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: No hurry. I will wait for your reply.

Comment (cquiroz) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: @soronpo it is very straightforward to add scala.js support after you merge #33. With the removal of `valueOf` it is much simpler. Feel free to merge and i'll make a new PR

Comment (soronpo) [inline shared/src/test/scala/singleton/ops/ImplicitConversionSpec.scala]: #33 merged. Feel free to submit an updated PR and I'll merge it.

**Comment by cquiroz** [comment]
This is ready for another review
